app/org/sagebionetworks/bridge/models/studies/Study.java

```diff
@@ -195,4 +196,19 @@
     
     /** @see #isHealthCodeExportEnabled(); */
     public void setHealthCodeExportEnabled(boolean enabled);
+    
+    /**
+     * Minimum supported app version number. If set, user app clients pointing to an older version will 
+     * fail with an httpResponse status code of 410.
+     */
+    public Integer getMinSupportedAppVersion(String osName);
+    
+    /** @see #getMinSupportedVersion(); */
+    public void setMinSupportedAppVersion(String osName, Integer version);
```

Comment:
I would remove this setter, it's jut more common in Java to to getMinSupportedAppVersions().put(osName, version);

In reality this will be set with the map setter both by DynamoDB and the researcher UI, all as a group.